pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaSourceConfig.java

```diff
@@ -97,6 +97,11 @@
             "The deserializer class for Kafka consumer to deserialize values. You typically shouldn't care this. "
                 + "Since the deserializer will be set by a specific implementation of `KafkaAbstractSource`.")
     private String valueDeserializationClass = "org.apache.kafka.common.serialization.ByteArrayDeserializer";
+    @FieldDoc(
+        defaultValue = "",
+        help =
+            "The consumer config properties file containing configs to be passed to Consumer.")
+    private String consumerConfigPropertiesFile;
```

Comment:
I am not sure how does this really work for connectors. because in connectors, it is very tricky to specify the properties file. I would suggest you using `private Map<String, String> producerProperties`, so that all the producer properties can be specified in the connector config file, rather than using a separate config file.

Reply:
In general, it is unnecessary to cache all the producer properties as kafka users prefer to pass user-defined properties in a separate config file or an additional option. Thus, it seems like a good choice to add a setting (e.g. producerConfigProperties) in the connector config file to pass extra properties.